mypy/main.py

```diff
@@ -264,26 +264,25 @@ def infer_python_version_and_executable(options: Options,
 
     # TODO: (ethanhs) Look at folding these checks and the site packages subprocess calls into
     # one subprocess call for speed.
-    if special_opts.python_executable is not None and special_opts.python_version is not None:
-        py_exe_ver = _python_version_from_executable(special_opts.python_executable)
-        if py_exe_ver != special_opts.python_version:
-            raise PythonExecutableInferenceError(
-                'Python version {} did not match executable {}, got version {}.'.format(
-                    special_opts.python_version, special_opts.python_executable, py_exe_ver
-                ))
-        else:
-            options.python_version = special_opts.python_version
-            options.python_executable = special_opts.python_executable
-    elif special_opts.python_executable is None and special_opts.python_version is not None:
-        options.python_version = special_opts.python_version
+
+    # Use the command line specified python_version/executable, or fall back to one set in the
+    # config file
+    python_version = special_opts.python_version or options.python_version
```

Comment:
I don't see how this can ever be None -- it is initialized to `sys.version_info[:2]` in options.py. So that means that the final `elif` below cannot be reached. What am I missing?

Reply:
Ah, good point. I suppose if we are falling back to using the Python version of the running Python that becomes unneeded.

Reply:
(Note that in general I'm not keen on this particular if/elif/elif approach to decoding combinations of two booleans -- see also #5855.)

Reply:
Yeah, I can definitely re-write this to be cleaner/more efficient. Thank you for pointing that out!